tex/tutorial.tex

```diff
@@ -0,0 +1,275 @@
+
+	\begin{lstlisting}
+	./compile.sh example_source_file.xtnd
+	\end{lstlisting}
+	% TODO: Incorporating C and Extend Standard Library/Initialization Functions?
+
+	\medskip \noindent
+	This should produce an \texttt{out} file. Running \texttt{./out} should successfully run the program.
+
+\section{Writing Extend Code - The Basics}
+Extend code can be written in a file that contains the conventional \texttt{.xtnd} extension. It consists of optional import statements and global variables, and one or more series of functions. Runnable Extend programs must contain a \texttt{main} function, and other functions can be written into the program as well.
+\newline \newline
+Below is a short tour of the features of Extend. More detail can be found in the next chapter - the Language Reference Manual. Please note Extend uses the \texttt{range} data type to represent multiple cells of data, composed of primitive data types or another \texttt{range}.
+
+	\subsection{Functions}
+	Functions are commonplace in Extend. They are declared with the syntax \texttt{function\_name([optional\_dimensions] function\_arguments)\{...\}}. Below is the syntax of the \texttt{main} function, which is needed to run Extend code, within a simple Extend program.
+
+	\begin{lstlisting}
+	main(args){
+			return 0;
+	}
+	\end{lstlisting}
+	% Perhaps a more informative program - but not too difficult.
+
+	\medskip \noindent
+ 	The return type of a function is a \textbf{range} of flexible size. \textbf{Ranges} will be discussed in a later section of this chapter. A function is composed of a series of statements, concluding with the return statement.
+	Note that the \textbf{return} statement is always the last statement in the function.
+
+		\subsubsection{Function Parameters}
+		Function parameters consist of zero or more ranges, signed with an optional dimension. If the arguments have been written with dimensions, those dimensions will be verified at runtime.
+		\begin{lstlisting}
+			foo([m,n] arg){
+				return m * n; \\ m and n initialized through arg1
+			}
+
+			bar([1,1] arg){
+				return 0; \\ 1 by 1 ranges should be primitive data types. If arg is not, a runtime error will be thrown
+			}
+		\end{lstlisting}
+
+	\subsection{Primitive Data Types}
+	Extend has three primitive data types: Numbers, Strings, and \texttt{empty}. An example of each is shown below.
+
+	\begin{lstlisting}
+		bar := 5; \\ Assigns bar to a number
+		foo := "Hello World"; \\ Assigns foo to a string
+		bar = empty; \\ Reassigns bar to empty
+	\end{lstlisting}
+
+	\medskip \noindent
+	\textbf{Note} that we declare a variable and assign it a value together with \texttt{:=}. If the variable already exists, \texttt{=} will suffice.
+	More information about Extend's primitive data types can be found in the Language Reference Manual.
+
+		\subsubsection{The typeof() function}
+		Extend offers a \texttt{typeof(expr)} function, which will return Number, String, Range, or Empty.
```

Comment:
As of now we still use type not typeof. If we should change LRM and FinalReport and Code should all change at the same time.